CMakeLists.txt

```
@@ -168,7 +168,7 @@ if (NOT BUILD_TESTS_ONLY)
     set(BUILD_UNIT_TESTS OFF)
   endif()
 
-  if (USE_EXTERNAL_MPI)
+  if (USE_EXTERNAL_MPI STREQUAL "ON")
```

My preference for this check would be:
```suggestion
  if (USE_EXTERNAL_MPI AND NOT USE_EXTERNAL_MPI STREQUAL "AUTO")
```

In general:
```cmake
if (NOT USE_EXTERNAL_MPI)
  # USE_EXTERNAL_MPI is a false constant
endif()

if (USE_EXTERNAL_MPI STREQUAL "AUTO")
  # USE_EXTERNAL_MPI is AUTO
endif()

if (USE_EXTERNAL_MPI)
  # USE_EXTERNAL_MPI is not a false constant
  # i.e. either a true constant or AUTO
endif()

if (USE_EXTERNAL_MPI AND NOT USE_EXTERNAL_MPI STREQUAL "AUTO")
  # USE_EXTERNAL_MPI is neither a false constant nor AUTO
  # i.e. it is a true constant
  # if this an especially common check, can add a local variable e.g.:
  set(USE_EXTERNAL_MPI_ON TRUE)
else()
  # USE_EXTERNAL_MPI_ON is false when USE_EXTERNAL_MPI is AUTO or a false constant
  set(USE_EXTERNAL_MPI_ON FALSE)
endif()
```

---

@omor1 question: does this line here not restrict the potential values of `USE_EXTERNAL_MPI` to `ON, OFF, and AUTO` 

https://github.com/ROCm/rocSHMEM/blob/develop/CMakeLists.txt#L77

If not, what is the precise meaning of that line?